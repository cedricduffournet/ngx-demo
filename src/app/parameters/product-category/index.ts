export * from './parameter-product-category.component';
