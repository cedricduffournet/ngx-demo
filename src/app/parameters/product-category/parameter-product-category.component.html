<app-product-category-list-view></app-product-category-list-view>
