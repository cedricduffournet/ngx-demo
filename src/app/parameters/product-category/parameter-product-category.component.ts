import { ChangeDetectionStrategy, Component, OnInit } from '@angular/core';

@Component({
  selector: 'app-parameter-product-category',
  templateUrl: './parameter-product-category.component.html',
  changeDetection: ChangeDetectionStrategy.OnPush
})
export class ParameterProductCategoryComponent implements OnInit {
  constructor() {}

  ngOnInit() {}
}
