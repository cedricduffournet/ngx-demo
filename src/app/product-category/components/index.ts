export * from './product-category-add/product-category-add.component';
export * from './product-category-update/product-category-update.component';
export * from './product-category-delete/product-category-delete.component';
export * from './product-category-form/product-category-form.component';
export * from './product-category-item/product-category-item.component';
export * from './product-category-items/product-category-items.component';
