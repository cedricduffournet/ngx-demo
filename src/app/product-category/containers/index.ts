export * from './product-category-add-modal/product-category-add-modal.component';
export * from './product-category-update-modal/product-category-update-modal.component';
export * from './product-category-delete-modal/product-category-delete-modal.component';
export * from './product-category-list-view/product-category-list-view.component';
