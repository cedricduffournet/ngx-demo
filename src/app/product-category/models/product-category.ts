import { schema } from 'normalizr';

export interface ProductCategory {
  id: number;
  name: string;
}

export const productCategorySchema = new schema.Entity('productCategories');
