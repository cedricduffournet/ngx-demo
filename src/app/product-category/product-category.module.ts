import { NgModule, ComponentFactoryResolver } from '@angular/core';

import { RouterModule } from '@angular/router';
import { EffectsModule } from '@ngrx/effects';
import { StoreModule } from '@ngrx/store';

import { SharedModule } from '@app/shared/shared.module';
import { ProductCategoryEffects } from '@app/product-category/state/effects';
import {
  ProductCategoryAddComponent,
  ProductCategoryUpdateComponent,
  ProductCategoryDeleteComponent,
  ProductCategoryFormComponent,
  ProductCategoryItemComponent,
  ProductCategoryItemsComponent
} from '@app/product-category/components';
import {
  ProductCategoryAddModalComponent,
  ProductCategoryUpdateModalComponent,
  ProductCategoryDeleteModalComponent,
  ProductCategoryListViewComponent
} from '@app/product-category/containers';
import { reducers } from '@app/product-category/state/reducers';
import { CoalescingComponentFactoryResolver } from '@app/coalescing-component-factory-resolver.service';

@NgModule({
  imports: [
    RouterModule,
    SharedModule,
    EffectsModule.forFeature([ProductCategoryEffects]),
    StoreModule.forFeature('productCategories', reducers)
  ],
  declarations: [
    ProductCategoryListViewComponent,
    ProductCategoryAddComponent,
    ProductCategoryUpdateComponent,
    ProductCategoryDeleteComponent,
    ProductCategoryFormComponent,
    ProductCategoryItemComponent,
    ProductCategoryItemsComponent,
    ProductCategoryAddModalComponent,
    ProductCategoryUpdateModalComponent,
    ProductCategoryDeleteModalComponent
  ],
  entryComponents: [
    ProductCategoryUpdateModalComponent,
    ProductCategoryAddModalComponent,
    ProductCategoryDeleteModalComponent
  ],
  exports: [ProductCategoryListViewComponent]
})
export class ProductCategoryModule {
  // see https://github.com/angular/angular/issues/14324
  constructor(
    coalescingResolver: CoalescingComponentFactoryResolver,
    localResolver: ComponentFactoryResolver
  ) {
    coalescingResolver.registerResolver(localResolver);
  }
}
