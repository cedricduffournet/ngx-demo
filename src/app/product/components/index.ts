export * from './product-add/product-add.component';
export * from './product-update/product-update.component';
export * from './product-delete/product-delete.component';
export * from './product-form/product-form.component';
export * from './product-item/product-item.component';
export * from './product-items/product-items.component';
export * from './product-details/product-details.component';
