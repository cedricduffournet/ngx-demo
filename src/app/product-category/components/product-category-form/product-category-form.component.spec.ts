import { ComponentFixture, TestBed } from '@angular/core/testing';

import { ReactiveFormsModule } from '@angular/forms';
import { TranslateModule } from '@ngx-translate/core';

import { FormModule } from '@app/shared/form';
import { ValidationActionModule } from '@app/shared/validation-action';
import { ProductCategoryFormComponent } from '@app/product-category/components';
import { ProductCategory } from '@app/product-category/models/product-category';

describe('ProductCategoryForm', () => {
  let fixture: ComponentFixture<ProductCategoryFormComponent>;
  let component: ProductCategoryFormComponent;

  beforeEach(() => {
    TestBed.configureTestingModule({
      imports: [
        ReactiveFormsModule,
        TranslateModule.forRoot(),
        FormModule,
        ValidationActionModule
      ],
      declarations: [ProductCategoryFormComponent]
    });

    fixture = TestBed.createComponent(ProductCategoryFormComponent);
    component = fixture.componentInstance;
  });

  it('should be created', () => {
    fixture.detectChanges();
    expect(fixture).toMatchSnapshot();
  });

  it('should not emit event if name not filled', () => {
    const productCategory = {
      name: ''
    } as ProductCategory;

    fixture.detectChanges();
    component.productCategoryForm.setValue(productCategory);

    spyOn(component.save, 'emit');
    component.onSubmit();

    expect(component.save.emit).not.toHaveBeenCalled();
  });


  it('should disable the form if processing', () => {
    component.processing = true;

    fixture.detectChanges();

    expect(fixture).toMatchSnapshot();
  });

  it('should emit save event with new productCategory, if the form is valid, when submitted', () => {
    const productCategory = {name: 'test' };
    fixture.detectChanges();

    component.productCategoryForm.setValue(productCategory);

    spyOn(component.save, 'emit');
    component.onSubmit();

    expect(component.save.emit).toHaveBeenCalledWith(productCategory);
  });

  it('should emit cancel event when click cancel button', () => {
    fixture.detectChanges();

    spyOn(component.cancel, 'emit');

    component.onCancel();

    expect(component.cancel.emit).toHaveBeenCalledWith('cancel');
  });

  it('should fill form with productCategory value', () => {
    const productCategory = {
      name: 'testName'
    } as ProductCategory;
    component.productCategory = productCategory;
    fixture.detectChanges();

    expect(component.productCategoryForm.value).toEqual(productCategory);
  });
});
