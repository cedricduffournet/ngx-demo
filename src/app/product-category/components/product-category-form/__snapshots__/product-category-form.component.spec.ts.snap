// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductCategoryForm should be created 1`] = `
<app-product-category-form
  cancel={[Function EventEmitter]}
  formBuilder={[Function FormBuilder]}
  processing="false"
  productCategory={[Function Object]}
  productCategoryForm={[Function FormGroup]}
  save={[Function EventEmitter]}
  submitted="false"
>
  <form
    class="ng-untouched ng-pristine ng-invalid"
    ng-reflect-form="[object Object]"
    novalidate=""
  >
    <section
      class="bg-white"
    >
      <div
        class="p-3"
      >
        <app-text-input
          class="ng-untouched ng-pristine ng-invalid"
          formcontrolname="name"
          ng-reflect-label="NAME"
          ng-reflect-name="name"
          ng-reflect-submitted="false"
        >
          <div
            class="form-group "
          >
            <label
              class="control-label"
            >
              NAME
            </label>
            <input
              class="form-control"
              placeholder=""
              type="text"
            />
            
            
            
            
            
            
          </div>
        </app-text-input>
      </div>
    </section>
    <app-wrapper-validation-action>
      <div
        class="d-flex justify-content-end align-items-center p-3"
      >
        <app-validation-button
          ng-reflect-confirm-disabled="false"
          ng-reflect-label-cancel="CANCEL"
          ng-reflect-label-confirm="SAVE"
        >
          <div
            class="button-validation-action text-center d-flex justify-content-end align-items-center p-3-group"
            ng-reflect-klass="button-validation-action text-"
            ng-reflect-ng-class=""
          >
            
            <button
              class="btn btn-cancel btn-lg btn-light"
              ng-reflect-klass="btn btn-cancel btn-lg"
              ng-reflect-ng-class="btn-light"
              type="button"
            >
               CANCEL 
            </button>
            
            <button
              class="btn btn-confirm btn-lg btn-primary"
              ng-reflect-klass="btn btn-confirm btn-lg"
              ng-reflect-ng-class="btn-primary"
              type="submit"
            >
              
              
              
              <span>
                SAVE
              </span>
            </button>
          </div>
        </app-validation-button>
      </div>
    </app-wrapper-validation-action>
  </form>
</app-product-category-form>
`;

exports[`ProductCategoryForm should disable the form if processing 1`] = `
<app-product-category-form
  cancel={[Function EventEmitter]}
  formBuilder={[Function FormBuilder]}
  processing={[Function Boolean]}
  productCategory={[Function Object]}
  productCategoryForm={[Function FormGroup]}
  save={[Function EventEmitter]}
  submitted="false"
>
  <form
    class="ng-untouched ng-pristine ng-invalid"
    ng-reflect-form="[object Object]"
    novalidate=""
  >
    <section
      class="bg-white"
    >
      <div
        class="p-3"
      >
        <app-text-input
          class="ng-untouched ng-pristine ng-invalid"
          formcontrolname="name"
          ng-reflect-label="NAME"
          ng-reflect-name="name"
          ng-reflect-submitted="false"
        >
          <div
            class="form-group "
          >
            <label
              class="control-label"
            >
              NAME
            </label>
            <input
              class="form-control"
              placeholder=""
              type="text"
            />
            
            
            
            
            
            
          </div>
        </app-text-input>
      </div>
    </section>
    <app-wrapper-validation-action>
      <div
        class="d-flex justify-content-end align-items-center p-3"
      >
        <app-validation-button
          ng-reflect-confirm-disabled="true"
          ng-reflect-label-cancel="CANCEL"
          ng-reflect-label-confirm="SAVE"
        >
          <div
            class="button-validation-action text-center d-flex justify-content-end align-items-center p-3-group"
            ng-reflect-klass="button-validation-action text-"
            ng-reflect-ng-class=""
          >
            
            
            <button
              class="btn btn-confirm btn-lg btn-primary"
              disabled=""
              ng-reflect-klass="btn btn-confirm btn-lg"
              ng-reflect-ng-class="btn-primary"
              type="submit"
            >
              
              <span
                class="spinner-border spinner-border-sm align-baseline"
                style="font-size: 1rem;"
              />
              
              <span>
                Â LOADING
              </span>
              
            </button>
          </div>
        </app-validation-button>
      </div>
    </app-wrapper-validation-action>
  </form>
</app-product-category-form>
`;
